libpldmresponder/test/libpldmresponder_fru_test.cpp

```diff
@@ -167,18 +174,32 @@ TEST(FruImpl, entityByObjectPath)
         bmcEntityTree(pldm_entity_association_tree_init(),
                       pldm_entity_association_tree_destroy);
 
-    InterfaceMap iface = {{"xyz.openbmc_project.Inventory.Item.Chassis", {}}};
+    sdbusplus::bus::bus bus(sdbusplus::bus::new_default());
+    TestInstanceIdDb instanceIdDb;
+    Requester requester(bus, "/abc/def", instanceIdDb);
+    auto event = sdeventplus::Event::get_default();
+
+    ObjectValueTree objects{
+        {sdbusplus::message::object_path(
+             "/xyz/openbmc_project/inventory/system"),
+         {{"xyz.openbmc_project.Inventory.Item.System", {}}}},
+        {sdbusplus::message::object_path(
+             "/xyz/openbmc_project/inventory/system/chassis"),
+         {{"xyz.openbmc_project.Inventory.Item.Chassis", {}}}},
+        {sdbusplus::message::object_path(
+             "/xyz/openbmc_project/inventory/system/chassis/motherboard"),
+         {{"xyz.openbmc_project.Inventory.Item.Board.Motherboard", {}}}}};
+
     pldm::responder::FruImpl mockedFruHandler(
         FRU_JSONS_DIR, "./fru_jsons/fru_master/fru_master.json", pdrRepo.get(),
-        entityTree.get(), bmcEntityTree.get(), nullptr);
+        entityTree.get(), bmcEntityTree.get(), nullptr, requester, nullptr, 0x9,
+        event, nullptr);
 
     // Good path
-    auto entityPtr = mockedFruHandler.getEntityByObjectPath(iface);
-    pldm_entity entity = *entityPtr;
+    auto entity = mockedFruHandler.getEntityByObjectPath(
+        objects, "/xyz/openbmc_project/inventory/system/chassis");
     EXPECT_TRUE(entity.entity_type == 45);
     // Bad Path
-    InterfaceMap invalidIface = {
-        {"xyz.openbmc_project.Inventory.Item.Motherboard", {}}};
-    entityPtr = mockedFruHandler.getEntityByObjectPath(invalidIface);
-    ASSERT_TRUE(!entityPtr);
```

Comment:
because getEntityByObjectPath is updated with another PR and it does not return the entity pointer but only the entity structure, there is test case to test if a structure is empty and the above test cannot be done with the new changes